services/gitea/gitea-base.js

```diff
@@ -0,0 +1,32 @@
+import { BaseJsonService } from '../index.js'
+
+export default class GiteaBase extends BaseJsonService {
+  static auth = {
+    passKey: 'gitea_token',
+    serviceKey: 'gitea',
+  }
+
+  async fetch({ url, options, schema, httpErrors }) {
```

Comment:
As discussed on Discord, I'd like to have some tests asserting that auth is being passed through correctly by this method before we merge this.

Reply:
I have attempted to add a test following the Stack Exchange example (current, prior to #9681). But am having difficult in getting it to even run in the basic state.

```
1) GiteaBase
       auth
         sends the auth information as configured:
     TypeError: Cannot read properties of undefined (reading 'services')
      at DummyGiteaService.invoke (file:///C:/Users/CanisHelix/IdeaProjects/Shields/core/base-service/base.js:430:36)
      at Context.<anonymous> (file:///C:/Users/CanisHelix/IdeaProjects/Shields/services/gitea/gitea-base.spec.js:32:33)
      at process.processImmediate (node:internal/timers:476:21)
```

I may need help on this auth test.

Reply:
Yep that's fine. As long as the test fails if we change the fake token, that's working as expected